File: app/src/main/java/cmsr/ipsacademy/net/api/controller.java
package cmsr.ipsacademy.net.api;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class controller
{
   private static final String url="http://172.16.10.89/android/";
   private static controller clienobject;
   private static Retrofit retrofit;

     controller()
     {
         retrofit=new Retrofit.Builder()
                      .baseUrl(url)
                      .addConverterFactory(GsonConverterFactory.create())
                      .build();
     }

     public static synchronized controller getInstance()
     {
          if(clienobject==null)
              clienobject=new controller();
          return  clienobject;
     }

    public apiset getapi()
     {
         return retrofit.create(apiset.class);
     }
}

File: app/src/main/java/cmsr/ipsacademy/net/api/controller.kt
package cmsr.ipsacademy.net.api


import cmsr.ipsacademy.net.Constants.BASE_URL
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object controller {

    fun getInstance(): Retrofit{
        return Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

}